test/fiddle/test_fiddle.rb

```diff
@@ -83,7 +83,7 @@ def test_dlopen_linker_script_group_linux
     handle = Fiddle.dlopen("libc.so")
     begin
       assert_equal("libc.so",
-                   File.basename(handle.file_name, ".*"))
+                   File.basename(handle.file_name, ".*")).gsub(/\.so.\d+/,'.so'))
```

Comment:
Let's use `gsub` for both of `libc.so.6` and `libc.so.6.1`:

```suggestion
                   File.basename(handle.file_name).gsub(/\.so(\.\d+)+\z/, ".so"))
```